src/main/java/projectbookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,35 @@
+package projectbookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstFieldName = constraintAnnotation.first();
+        secondFieldName = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        Class<?> clazz = value.getClass();
+        try {
+            Field firstField = clazz.getDeclaredField(firstFieldName);
+            firstField.setAccessible(true);
+            Object firstObj = firstField.get(value);
```

Comment:
looks a bit cleaner, same for the `secondObj`
```suggestion
            Object firstFieldValue = firstField.get(value);
```